sqlalchemy_redshift/ddl.py

```diff
@@ -0,0 +1,260 @@
+import sqlalchemy as sa
```

Comment:
Can you tell me a bit more about the decision to break out a `ddl.py`? Does the PostgreSQL dialect or other dialects do this?

Reply:
My rough understanding is that `DDL` ~= `Create/Drop` clauses.  In this spirit, I think they are different from the `_ExecutableClauses` found in `commands.py`.  This is my main reason for splitting; its pretty pedantic.  I can definitely see the reasoning behind keeping it all in one `commands.py` file for simplicity.  

As to leave no stone unturned, I've looked closer and found that `ddl.py` has its own module under `sqlalchemy/sql/ddl.py`, whereas query clauses seem to be mostly in `sqlalchemy/sql/selectable.py`.

Once we go into the dialect tho, `sqlalchemy/dialects/postgresql`, its the `base.py` where most of the dropping and creating happens (i.e. contains the DDL modifications for PostgreSQL).  Under  `sqlalchemy/dialects/postgresql`, there seem to be special modules for some of PostgreSQL's special types (json, array, etc.).

In both cases, I fail to find the direct analogy of Redshift's "commands" concept.  However, I think that `commands` is apt for describing `COPY`, `UNLOAD`, and the other existing function in `commands.py`.  I'm pretty new to this level of `sqlalchemy` tho, so I could just be missing something.

In summary, we can either:
 * separate out the concept of Redshift-specific "creation" commands to `ddl` vs.
 * keep it all in `commands`, just to keep it simple.

happy to take your lead, I don't mind spending the time for option 2 if you think its best.

Reply:
Thanks for the research here. I suppose it comes down to what makes most sense for the user when importing. Should we prefer:

     from sqlalchemy_redshift.ddl import CreateMaterializedView

Or:

     from sqlalchemy_redshift import CreateMaterializedView

Having it under `ddl` seems reasonable. I agree with you that DDL statements seem different from commands, so separating them is a good idea. And I think I also like the idea that we keep "extras" like these outside of the main `dialect.py` filed.

Reply:
well right now, I cannot import `CopyCommand` from `sqlalchemy_redshift`, but I can import it from `sqlalchemy_redshift.dialect` or `sqlalchemy_redshift.commands`.

If `CreateMaterializedView` lives in `ddl.py`, then analogously it can be imported from `sqlalchemy_redshift.dialect` or `sqlalchemy_redshift.ddl`.  This is because I followed the example of the package, where `dialect.__all__` has all commands and types.  

This follows the pattern in `sqlalchemy/dialects/postgresql/__init__.py`, where `__all__` contains elements like `BIGINT` and `CreateEnumType`.  The import pattern in SQLAlchemy seems to be like:
```python
from sqlalchemy.dialects.postgresql import X
```

perhaps then, something like
```python
from sqlalchemy_redshift.dialect import X
```
is not too different.

My proposal is then: 
 * we can point the users to `from sqlalchemy_redshift.dialect import X`.
 * If this package grows, and we want to factor out a `base.py`, etc, `dialect.py` can become a (depreciatable) high-level namespace that contains all the dialect elements, much like `sqlalchemy/dialects/postgresql/__init__.py`.

Reply:
I appreciate the thought you've put into this. I leave it up to you to decide what the "preferred" import should look like (ddl vs. dialect), and make sure that the docstring example and the tests import it using the preferred style.

Let me know when you consider this done, and I'll run a final integration test and merge.

Reply:
okay, I'm done, I changed any `commands` or `ddl` imports in the tests to the `dialects` namespace.  I likewise updated the docs that are in the docstrings to reflect this change as well.